tensorflow/compiler/mlir/xla/transforms/legalize_tf.cc

```diff
@@ -1446,6 +1446,120 @@ class ConvertRollOp : public OpRewritePattern<TF::RollOp> {
   }
 };
 
+/// Converts a TF::SeluOp to HLO.
+/// Selu(x) = scaledAlpha * (exp(x) - 1) if x < 0 else scale * x. Here
+/// scale(`kScale`) is taken as a constant 1.0507009873554804934193349852946 and
+/// scaledAlpha(`kScaledAlpha`) is taken as a
+/// constant 1.7580993408473768599402175208123. This choice of constants is in
+/// accordance with existing TF2XLA bridge.
+class ConvertSeluOp : public OpRewritePattern<TF::SeluOp> {
```

Comment:
We should move this to lower_tf.cc instead of directly lowering to mhlo. That way the pattern is useful in other contexts as well.

I think it should have been possible to implement these patterns using tablegen. Did you consider that?

If you are in a hurry, we can add TODO for these now.

Reply:
Hi @smit-hinsu I have added these as DRRs. Please re-review.

Reply:
Did you consider moving the pattern to lower_tf so that other backends other than mhlo can make use of the pattern?
https://github.com/tensorflow/tensorflow/blob/c83cdf85c5937730af51a384c0cc605bfb8f9214/tensorflow/compiler/mlir/tensorflow/transforms/lower_tf.cc

This could also simplify the pattern by making use of other TensorFlow ops.

This shouldn't be difficult but if you need to commit this sooner for some deadline, we can add a TODO and I can review this.

Reply:
@smit-hinsu I have started work on this. Will update the PR soon. Thanks!